### PR TITLE
Implementation of the metrics role with PCP, Redis and Grafana.

### PR DESCRIPTION
**Pull request by natoscott**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,177 +1,19 @@
-
```
Comment (richm) [inline LICENSE:1]: Is there anything left of the original code which was licensed under the APL?

File: README.md [hunk 2]
```diff
@@ -1,39 +1,88 @@
...
 
-### Configure metrics
-In order to run this role you will need set the following variables:
+Uses features of Performance Co-Pilot (PCP) v5+, Grafana v6+, and Redis v5+.
 
-- `env_name:` (required - default: `"engine"`)
+## Role Variables
 
-  Environment name. Is used to identify the source of the collected data.
+    monitored_hosts: []
```
Comment (richm) [inline README.md:13]: variable names should begin with the role name e.g. `metrics_monitored_hosts` - see https://github.com/oasis-roles/meta_standards#naming-things

File: README.md [hunk 3]
```diff
@@ -1,39 +1,88 @@
...
+Boolean flag allowing host to be setup with graphing services.
+Enabling this starts PCP and grafana servers for visualizing PCP metrics.
 
-In order to set these variable add the required variables to the config.yml
-or in the command line.
+    with_query_service: false
 
-You don't need to update the configuration file if you wish to use default options.
+Boolean flag allowing host to be setup with time series query services.
+Enabling this starts PCP and redis servers for querying recorded PCP metrics.
```
Comment (richm) [inline README.md:33]: `retention_days` (or `metrics_retention_days`) is a variable that needs to be documented?

File: meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  #   versions:
+  #   - all
+  #   - 1.0
+  #   - 7
+  #   - 99.99
+  platforms:
+    # Replace the below with your platform list:
+    - name: Fedora
+      versions:
+        - all
```
Comment (richm) [inline meta/main.yml:54]: Please use specific versions - e.g. 30, 31, 32

File: meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    - name: EL
+      versions:
+        - all
```
Comment (richm) [inline meta/main.yml:57]: please use specific versions - 7, 8 (6?)

File: molecule/default/molecule.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: MIT
+---
+dependency:
+  name: galaxy
+driver:
+  name: docker
+lint: |
+  yamllint -c .yamllint.yml .
```
Comment (richm) [inline molecule/default/molecule.yml:8]: we don't currently support molecule v3

File: molecule/default/molecule.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  ansible-lint
+  flake8
+platforms:
+  - name: centos-6
+    image: docker.io/linuxsystemroles/centos-6
+    privileged: true
```
Comment (richm) [inline molecule/default/molecule.yml:14]: You can remove this if you do not support el6

File: tasks/main.yml [hunk 8]
```diff
@@ -1,9 +1,10 @@
...
-    name: "metrics/roles/initial-validations"
+# SPDX-License-Identifier: MIT
+---
 
-- name: Run collectd sub-role
-  include_role:
-    name: "metrics/roles/collectd"
-  when:
-    - failed_validation is undefined
+#- include_tasks: pcp.yml
```
Comment (richm) [inline tasks/main.yml:4]: this is commented out?

File: meta/main.yml [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+  galaxy_tags: []
+  # List tags for your role here, one per line. A tag is a keyword that
+  # describes and categorizes the role. Users find roles by searching for tags.
+  # Be sure to remove the '[]' above, if you add tags to this list.
+  #
+  # NOTE: A tag is limited to a single word comprised of alphanumeric
+  #       characters. Maximum 20 tags per role.
+
+dependencies: []
```
Comment (richm) [inline meta/main.yml:67]: looks like there is at least 1 dependency on `performancecopilot.pcp` ?

File: README.md [hunk 10]
```diff
@@ -1,39 +1,88 @@
...
 
-License
--------
+## Dependencies
```
Comment (richm) [inline README.md:35]: performancecopilot.pcp ?  grafana?  redis?  others?

File: meta/main.yml (hunk 4 above)
Comment (natoscott) [reply meta/main.yml:54]: Hmm, really?  Its all - I'd rather not have to update this for each new Fedora version...?

File: meta/main.yml (hunk 5 above)
Comment (natoscott) [reply meta/main.yml:57]: OK, this one's less invasive - will do.  (and will add el6, although some newer functionality will be missing on older versions - is that frowned on?  Or should I just exclude the old versions?  even el7 has some missing features of this role under el8)

File: meta/main.yml (hunk 4 above)
Comment (richm) [reply meta/main.yml:54]: > Hmm, really? Its all - I'd rather not have to update this for each new Fedora version...?

We'll have to figure out some way to automate it, but yes, we have to explicitly configure these in our CI.

File: meta/main.yml (hunk 5 above)
Comment (richm) [reply meta/main.yml:57]: > OK, this one's less invasive - will do. (and will add el6, although some newer functionality will be missing on older versions - is that frowned on? Or should I just exclude the old versions? even el7 has some missing features of this role under el8)

This is the standard way to handle setting variables specific to a platform/version - https://github.com/linux-system-roles/kernel_settings/pull/7/files#diff-0363d71adca523f53e6765d01762b5c0R44 - if you need to perform platform/version specific tasks you can do the same thing with tasks files e.g. `tasks/setup_{{ ansible_distribution }}_{{ ansible_distribution_version }}.yml`

File: tasks/main.yml [hunk 11]
```diff
@@ -4,7 +4,7 @@
 - include_tasks: pcp.yml
 
 - include_tasks: redis.yml
-  when: metrics_query_service == true
+  when: metrics_query_service == 'true'
```
Comment (richm) [inline tasks/main.yml:7]: If `metrics_query_service` is a boolean valued variable, I suggest using
```yaml
when: metrics_query_service | bool
```
instead

File: tasks/main.yml [hunk 12]
```diff
@@ -1,9 +1,10 @@
...
-    name: "metrics/roles/collectd"
-  when:
-    - failed_validation is undefined
+- include_tasks: pcp.yml
+
+- include_tasks: redis.yml
+  when: metrics_query_service == 'true'
+
+- include_tasks: grafana.yml
+  when: metrics_graph_service == 'true'
```
Comment (richm) [inline tasks/main.yml:10]: same as above - use `when: metrics_graph_service | bool`

File: README.md [hunk 13]
```diff
@@ -1,39 +1,93 @@
...
+
+Centralized metric recording setup for several remote hosts and
+scalable metric recording, analysis and visualization setup for
+the local host, providing a REST API server with an OpenMetrics
+endpoint, graphs and scalable querying.
+
+```yaml
+- hosts: monitors
+  vars:
+    metrics_monitored_hosts: [app.acme.com, db.acme.com, nas.acme.com]
```
Comment (pcahyna) [inline README.md:78]: please use RFC 2606 domain names (example.com, ...) in examples

File: tasks/main.yml (hunk 12 above)
Comment (pcahyna) [inline tasks/main.yml:10]: please do not compare booleans to `'true'`. Use the `| bool` filter instead which does a proper conversion of user-supplied boolean values.

**Pull request merged**